### PR TITLE
filter out ts keywords in js file

### LINKED ISSUE #29212: In JS, completion list suggests TypeScript syntax
**TypeScript Version:**  3.3.0-dev.20181222

**Search Terms:**

**Code**

```ts
// trigger completion list anywhere in a JS file
```

**Expected behavior:**

Completion list doesn't contain TypeScript specific syntax in JS files.

**Actual behavior:**

In JS files it suggests `enum`, `interface`, `namespace`, `type`, `declare`, `abstract`, `readonly`, `public`, `protected`, `private`.

**Playground Link:** 

**Related Issues:**

### PR DESCRIPTION
**Pull request by fuafa**
Fixes #29212

Need help, the public APIs test fails even in a fresh workspace
```
  1) Public APIs
       for the language server
         should be acknowledged when they change:
     Error: The baseline file api/tsserverlibrary.d.ts has changed.
      at writeComparison (src/harness/harness.ts:2079:23)
      at Object.runBaseline (src/harness/harness.ts:2089:13)
      at Context.<anonymous> (src/testRunner/unittests/publicApi.ts:13:30)
```

### CODE REVIEW

File: src/services/completions.ts [hunk 1]
```diff
@@ -127,6 +128,27 @@ namespace ts.Completions {
         return { isGlobalCompletion: isInSnippetScope, isMemberCompletion: isMemberCompletionKind(completionKind), isNewIdentifierLocation, entries };
     }
 
+    const tsKeywords: string[] = [
+        SyntaxKind.EnumKeyword,
+        SyntaxKind.AbstractKeyword,
+        SyntaxKind.AsKeyword,
```
Comment (sheetalkamat) [inline src/services/completions.ts]: I think these need to be context sensitive and need to be paired with different keyword filters. 
Eg. `as` is valid input `import * /*here*/as name from "module-name";`

Comment (fuafa) [reply src/services/completions.ts]: I think they are still context sensitive as usual, as showed in the test cases. the `as` keyword should be valid in JS, I removed it. And I updated the test cases to be more precise.

File: src/services/completions.ts [hunk 2]
```diff
@@ -127,6 +128,26 @@ namespace ts.Completions {
...
+        SyntaxKind.InferKeyword,
+        SyntaxKind.IsKeyword,
+        SyntaxKind.KeyOfKeyword,
+        SyntaxKind.ModuleKeyword,
+        SyntaxKind.NamespaceKeyword,
+        SyntaxKind.NeverKeyword,
+        SyntaxKind.ReadonlyKeyword,
+        SyntaxKind.TypeKeyword,
+        SyntaxKind.UniqueKeyword,
+        SyntaxKind.UnknownKeyword
```
Comment (DanielRosenwasser) [inline src/services/completions.ts]: Does this mean you don't get completions for `any` and `unknown` and `never`, etc. in JSDoc?

Comment (sheetalkamat) [reply src/services/completions.ts]: And that's why it needs to be context sensitive.

Comment (fuafa) [reply src/services/completions.ts]: Oops, forget this situation, fixed.

File: src/services/completions.ts [hunk 3]
```diff
@@ -127,6 +128,26 @@ namespace ts.Completions {
...
+    ].map(tokenToString) as string[];
```
[the remark this answers is not in the document]
Comment (fuafa) [reply src/services/completions.ts]: Fixed

File: src/services/completions.ts [hunk 4]
```diff
@@ -113,7 +113,10 @@ namespace ts.Completions {
 
         if (keywordFilters !== KeywordCompletionFilters.None) {
             const entryNames = arrayToSet(entries, e => e.name);
-            for (const keywordEntry of getKeywordCompletions(keywordFilters)) {
+            const keywordsCompletion = isSourceFileJS(sourceFile) && keywordFilters !== KeywordCompletionFilters.TypeKeywords
```
Comment (sheetalkamat) [inline src/services/completions.ts]: Can we create separate filters in all places that set keyword filters so they are handled now and later correctly. It should be done by checking isSourceFileJS and setting new keywordFilter eg.: `ClassElementKeywordsJS` etc.

Comment (fuafa) [reply src/services/completions.ts]: Yes, please check am I doing it right @sheetalkamat 😄

**Pull request closed without merging**